pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +120,254 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
+					return reconcile.Result{}, statusError
+				}
+
+				err = NewSignupError("No target clusters available")
+				return reconcile.Result{}, err
+			}
+		}
+
+		// Provision the MasterUserRecord
+		err = r.provisionMasterUserRecord(instance, targetCluster, reqLogger)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		return reconcile.Result{}, r.updateStatus(reqLogger, instance, r.setStatusPendingApproval, "")
+	}
+
 	return reconcile.Result{}, nil
 }
+
+// provisionMasterUserRecord does the work of provisioning the MasterUserRecord
+func (r *ReconcileUserSignup) provisionMasterUserRecord(userSignup *toolchainv1alpha1.UserSignup, targetCluster string, logger logr.Logger) error {
+	userAccounts := []toolchainv1alpha1.UserAccountEmbedded{
+		{
+			TargetCluster: targetCluster,
+			Spec: toolchainv1alpha1.UserAccountSpec{
+				UserID: userSignup.Spec.UserID,
+				NSLimit: "default",
+				NSTemplateSet: toolchainv1alpha1.NSTemplateSetSpec{
+					Namespaces: []toolchainv1alpha1.Namespace{},
+				},
+			},
+		},
+	}
+
+	// TODO Update the MasterUserRecord with NSTemplateTier values
+	// SEE https://jira.coreos.com/browse/CRT-74
+
+	mur := &toolchainv1alpha1.MasterUserRecord{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: userSignup.Name,
+			Namespace: userSignup.Namespace,
+		},
+		Spec: toolchainv1alpha1.MasterUserRecordSpec{
+			UserID: userSignup.Spec.UserID,
+			UserAccounts: userAccounts,
+		},
+	}
+
+	err := controllerutil.SetControllerReference(userSignup, mur, r.scheme)
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(logger, userSignup, r.setStatusFailedToCreateMUR, err,
+			"Error setting controller reference for MasterUserRecord %s", mur.Name)
+	}
+
+	err = r.client.Create(context.TODO(), mur)
+	if err != nil {
+		// If the MasterUserRecord already exists (for whatever strange reason), we simply set the status to complete
+		if errors.IsAlreadyExists(err) {
+			logger.Info("MasterUserRecord already exists, setting status to Complete")
+			return r.updateStatus(logger, userSignup, r.setStatusComplete, "")
+
+		} else {
+			return r.wrapErrorWithStatusUpdate(logger, userSignup, r.setStatusFailedToCreateMUR, err,
+				"Error creating MasterUserRecord")
+		}
+	}
+
+	logger.Info("Created MasterUserRecord", "Name", userSignup.Name, "TargetCluster", targetCluster)
+	return nil
+}
+
+// ReadUserApprovalPolicyConfig reads the ConfigMap for the toolchain configuration in the operator namespace, and returns
+// the config map value for the user approval policy (which will either be "manual" or "automatic")
+func (r *ReconcileUserSignup) ReadUserApprovalPolicyConfig(namespace string) (string, error) {
+	cm, err := r.clientset.CoreV1().ConfigMaps(namespace).Get(config.ToolchainConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return config.UserApprovalPolicyManual, nil
+		}
+		return "", err
+	}
+
+	return cm.Data[config.ToolchainConfigMapUserApprovalPolicy], nil
```

Comment:
Looks like you are trying to return map's value with key `config.ToolchainConfigMapUserApprovalPolicy`, but if somehow, user forget to add this `key: value` pair in configmap, then we'll get this as empty and we are not handling this empty case in reconcile loop. 

We should do like
```
val, ok := cm.Data[config.ToolchainConfigMapUserApprovalPolicy]
if  !ok {
    return default_val, nil 
} else {
    return val, nil
}
```
or we should handle this empty case in reconcile loop

empty map  demo - https://play.golang.org/p/XgS8Flyh8x_r